composer/trainer/deepspeed.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from dataclasses import dataclass
+
+import yahp as hp
+
+
+@dataclass
+class DeepSpeedHparams(hp.Hparams):
+    """Params for configuration of DeepSpeed."""
+
+    enabled: bool = hp.required("Whether to use DeepSpeed.")
```

Comment:
Why do we need the enabled flag? If the params are specified in the trainer, then wouldn't it be enabled?

Reply:
Or, as an alternative, maybe just add a `use_deepspeed` flag in the trainer hparams if there are hparams?

Reply:
When I add support for things like ZeRO, there's going to be a lot more DeepSpeed-relevant hparams, so I definitely at least want this object as a way of organizing those hparams.